server/controllers/wishlistController.js

```diff
@@ -0,0 +1,202 @@
+const Wishlist = require('../models/wishlist');
+
+const _getWishlistCost = list => {
+  // Accept list as Array of objects
+  // Should return a single cost for the entire wishlist
+  let totalCost = 0;
+
+  list.forEach(item => {
+    totalCost += item.itemCost * item.itemQuantity;
+  });
+
+  return totalCost;
+};
+
+exports.addItem = (req, res) => {
+  // Bad request if no body is sent
+  if (Object.keys(req.body).length === 0) {
+    return res.status(400).json({
+      Error: 'Body missing data',
+    });
+  }
+
+  const familyId = req.params.familyId;
+  const item = req.body.item;
+
+  // TODO: Check structure of item
+  // Prevent duplicate items
+
+  try {
+    Wishlist.find({ family: familyId }).exec((err, wishlist) => {
+      if (err) {
+        res.status(500).json({ Error: err });
+      }
+
+      wishlist[0].list.push(item);
+
+      wishlist[0].totalListCost = _getWishlistCost(wishlist[0].list);
+
+      wishlist[0].save((err, wishlist) => {
+        if (err) {
+          res.status(500).json({ Error: err });
+        }
+
+        res.status(201).json({
```

Comment:
You should be returning the newly created object in the response.
From MDN: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201
The HTTP 201 Created success status response code indicates that the request has succeeded and has led to the creation of a resource. **The new resource is effectively created before this response is sent back. and the new resource is returned in the body of the message**, its location being either the URL of the request, or the content of the Location header